core/src/main/java/org/apache/iceberg/BaseTransaction.java

```diff
@@ -404,6 +397,9 @@ private void commitSimpleTransaction() {
           .onFailure((file, exc) -> LOG.warn("Failed to delete uncommitted file: {}", file, exc))
           .run(ops.io()::deleteFile);
 
+      if (e instanceof ValidationFailureException) {
```

Comment:
Looks like this is here because you want to share the cleanup code, but I think it would be cleaner to move that to a method that gets called from separate `catch` blocks.

Reply:
yeah. It is cleaner that way. Fixed now.